File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/configurequeueemail1.cs
// <snippetconfigurequeueemail1>


 // Define some anonymous types to define the range 
 // of possible queue property values.
 var IncomingEmailDeliveryMethods = new
 {
     None = 0,
     EmailRouter = 2,
     ForwardMailbox = 3
 };

 var IncomingEmailFilteringMethods = new
 {
     AllEmailMessages = 0,
     EmailMessagesInResponseToCrmEmail = 1,
     EmailMessagesFromCrmLeadsContactsAndAccounts = 2
 };

 var OutgoingEmailDeliveryMethods = new
 {
     None = 0,
     EmailRouter = 2
 };

 // Update a queue instance and set its email property values.
 Queue configureQueue = new Queue()
 {
     QueueId = _queueId,
     IncomingEmailDeliveryMethod = new OptionSetValue(
         IncomingEmailDeliveryMethods.EmailRouter),
     IncomingEmailFilteringMethod = new OptionSetValue(
         IncomingEmailFilteringMethods.EmailMessagesInResponseToCrmEmail),
     OutgoingEmailDeliveryMethod = new OptionSetValue(
         OutgoingEmailDeliveryMethods.None),

     // TODO: Update with appropriate address 
     // for accessing the e-mail router. 
     EMailAddress = "someone@example.com",                        
 };

 _serviceProxy.Update(configureQueue);

// </snippetconfigurequeueemail1>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/businessmanagementvb/vb/configurequeueemail1.vb
' <snippetconfigurequeueemail1>


 ' Define some anonymous types to define the range 
 ' of possible queue property values.
 Dim IncomingEmailDeliveryMethods = New With {Key .None = 0, Key .EmailRouter = 2, Key .ForwardMailbox = 3}

 Dim IncomingEmailFilteringMethods = New With {Key .AllEmailMessages = 0, Key .EmailMessagesInResponseToCrmEmail = 1, _
                                               Key .EmailMessagesFromCrmLeadsContactsAndAccounts = 2}

 Dim OutgoingEmailDeliveryMethods = New With {Key .None = 0, Key .EmailRouter = 2}

 ' Update a queue instance and set its email property values.
 Dim configureQueue As New Queue() With { _
     .QueueId = _queueId, .IncomingEmailDeliveryMethod = New OptionSetValue(IncomingEmailDeliveryMethods.EmailRouter), _
     .IncomingEmailFilteringMethod = New OptionSetValue(IncomingEmailFilteringMethods.EmailMessagesInResponseToCrmEmail), _
     .OutgoingEmailDeliveryMethod = New OptionSetValue(OutgoingEmailDeliveryMethods.None), _
     .EMailAddress = "someone@example.com", .EmailUsername = "exampleuser", .EmailPassword = "examplepassword"}
     ' TODO: Update with appropriate credentials 
     ' for accessing the e-mail router. 

 _serviceProxy.Update(configureQueue)

' </snippetconfigurequeueemail1>